frontend/src/api/k8s/__tests__/notebooks.spec.ts

```diff
@@ -0,0 +1,936 @@
+      type: 'remove',
+      settings: [] as Toleration[],
+    } as TolerationChanges;
+    const enablePipelines = true;
+    const mockNotebook = mockNotebookK8sResource({ uid });
+
+    k8sPatchResourceMock.mockResolvedValue(mockNotebookK8sResource({ uid }));
+
+    const renderResult = await startNotebook(mockNotebook, tolerationChanges, enablePipelines);
+
+    expect(k8sPatchResourceMock).toHaveBeenCalledWith({
+      model: NotebookModel,
+      queryOptions: { name: mockNotebook.metadata.name, ns: mockNotebook.metadata.namespace },
+      patches: [
+        startPatch,
+        getTolerationPatch(tolerationChanges),
+        getPipelineVolumePatch(),
+        getPipelineVolumeMountPatch(),
+      ],
+    });
+    expect(createElyraServiceAccountRoleBinding).toHaveBeenCalledWith(mockNotebook);
+    expect(k8sPatchResourceMock).toHaveBeenCalledTimes(1);
+    expect(renderResult).toStrictEqual(mockNotebook);
+  });
+  it('should start a notebook without pipelines', async () => {
+    const tolerationChanges = {
+      type: 'nothing',
+      settings: [] as Toleration[],
+    } as TolerationChanges;
+    const enablePipelines = false;
+
+    k8sPatchResourceMock.mockResolvedValue(mockNotebookK8sResource({ uid }));
+    const mockNotebook = mockNotebookK8sResource({ uid });
+
+    const renderResult = await startNotebook(mockNotebook, tolerationChanges, enablePipelines);
+
+    expect(k8sPatchResourceMock).toHaveBeenCalledWith({
+      model: NotebookModel,
+      patches: [startPatch],
+      queryOptions: { name: mockNotebook.metadata.name, ns: mockNotebook.metadata.namespace },
+    });
+
+    expect(k8sPatchResourceMock).toHaveBeenCalledTimes(1);
+    expect(createElyraServiceAccountRoleBinding).not.toHaveBeenCalled();
+    expect(renderResult).toStrictEqual(mockNotebookK8sResource({ uid }));
+  });
+  it('should handle errors and rethrow', async () => {
+    const tolerationChanges = {
+      type: 'add',
+      settings: [] as Toleration[],
+    } as TolerationChanges;
+    const enablePipelines = true;
+    const mockNotebook = mockNotebookK8sResource({ uid });
+
+    k8sPatchResourceMock.mockRejectedValue(new Error('error1'));
+    await expect(startNotebook(mockNotebook, tolerationChanges, enablePipelines)).rejects.toThrow(
+      'error1',
+    );
+    expect(k8sPatchResourceMock).toHaveBeenCalledTimes(1);
+    expect(k8sPatchResourceMock).toHaveBeenCalledWith({
+      model: NotebookModel,
+      queryOptions: { name: mockNotebook.metadata.name, ns: mockNotebook.metadata.namespace },
+      patches: [
+        startPatch,
+        getTolerationPatch(tolerationChanges),
+        getPipelineVolumePatch(),
+        getPipelineVolumeMountPatch(),
+      ],
+    });
+  });
+});
+describe('stopNotebook', () => {
+  it('should stop a notebook', async () => {
+    const name = 'test-notebook';
+    const namespace = 'test-project';
+
+    k8sPatchResourceMock.mockResolvedValue(mockNotebookK8sResource({ uid }));
```

Comment:
Either assign these values into the mock, or pull them from the mock since the mock has values for name and namespace already.
```suggestion
    const name = 'test-notebook';
    const namespace = 'test-project';

    k8sPatchResourceMock.mockResolvedValue(mockNotebookK8sResource({ uid, name, namespace }));
```